components/public-api-server/pkg/oidc/service.go

```diff
@@ -313,9 +313,15 @@ func (s *Service) createSession(ctx context.Context, flowResult *AuthFlowResult,
 	}
 
 	url := fmt.Sprintf("http://%s/session", s.sessionServiceAddress)
-	res, err := http.Post(url, "application/json", bytes.NewReader(payload))
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, url, bytes.NewReader(payload))
```

Comment:
drive-by change to make sure we make the request with the context, for context propagation